tools/applicationinsights-web-snippet/src/snippet.ts

```diff
@@ -202,11 +204,59 @@ declare var cfg:ISnippetConfig;
             "js2.cdn.applicationinsights.io",
             "js2.cdn.monitor.azure.com",
             "az416426.vo.msecnd.net" // this domain is supported but not recommended
-        ]
-    
-        // Assigning these to local variables allows them to be minified to save space:
-        let targetSrc : string = (aiConfig as any)["url"] || cfg.src
-        if (targetSrc) {
+        ];
+
+        let targetSrc : string = (aiConfig as any)["url"] || cfg.src;
+        const fallback = () => setScript(targetSrc, null);
+        if (cfg.sri) {
+            const match = targetSrc.match(/^((http[s]?:\/\/.*\/)([\w]+\.))?(\d+(\.\d+){0,2})\.(([\w]+\.){0,2}js)$/);
+            
+            if (match && match.length === 8) {
+                const integrityUrl = `${match[1]}${match[4]}.integrity.json`;
```

Comment:
This doesn't look correct...
for http://js.monitor.azure.com/scripts/b/ai.3.min.js
match[1] looks like `http://js.monitor.azure.com/scripts/b/ai.` and
match[4] looks like ""

So while this "works" (because match[4] is empty) I think it should just be `match[1] + ".integrity.json"` (note not using the string replacement "`${match[1]....." for generated code sizing.

Reply:
work for local test: 
http://js.monitor.azure.com/scripts/b/ai.3.min.js
http://js.monitor.azu/sdfsd2344f.3.js
http://azure.com/script/b/ai.3.2.1.min.js
http://azure.com/script/b/ai.3.js
http://azure.com/script/b/ai444.3.js
http://azure.com/script/b/ai444ddf.3.js
http://js.monitor.azure.com/scripts/b/ai.3.2.1.gbl.min.js